File: haskell/beginner/uri2850/uri2850.hs
module Main where

papagaio :: [Char] -> [Char]
papagaio s
  | s == "esquerda" = "ingles"
  | s == "direita" = "frances"
  | s == "nenhuma" = "portugues"
  | otherwise = "caiu"

main :: IO ()
main = getContents >>= mapM_ (putStrLn . papagaio) . lines

File: c/beginner/uri2850/uri2850.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char *saida(char *papagaio) {
  if (!strcmp(papagaio, "esquerda"))
    return "ingles";
  if (!strcmp(papagaio, "direita"))
    return "frances";
  if (!strcmp(papagaio, "nenhuma"))
    return "portugues";
  if (!strcmp(papagaio, "as duas"))
    return "caiu";
  return "";
}

int main(void) {
  char *papagaio = (char *)calloc(9, sizeof(char));

  while (scanf("%[^\n]\n", papagaio) == 1)
    printf("%s\n", saida(papagaio));

  return EXIT_SUCCESS;
}